app/javascript/packs/foremCreatorSignup.js

```diff
@@ -0,0 +1,98 @@
+function setDefaultUsername(event) {
+  if (
+    document
+      .getElementsByClassName('js-creator-signup-username-row')[0]
+      .classList.contains('hidden')
+  ) {
+    const name = event.target.value;
+    // It's the first user and so we can assume that this username is not taken.
+    const usernameHint = createUsernameHint(name);
+    setUsernameHint(usernameHint);
+    setUsernameField(usernameHint);
+    showHintRow();
+  }
+}
+
+function createUsernameHint(name) {
+  return name
+    .toLowerCase()
+    .replace(/[^a-zA-Z0-9]/g, '_')
+    .substr(0, 30);
+}
+
+function showHintRow() {
+  const hintRow = document.getElementsByClassName(
+    'js-creator-signup-username-hint-row',
+  )[0];
+  hintRow.classList.remove('hidden');
+}
+
+function setUsernameHint(usernameHint) {
+  const usernameHintDisplay = document.getElementsByClassName(
+    'js-creator-signup-username-hint',
+  )[0];
+  usernameHintDisplay.innerHTML = usernameHint;
+}
+
+function setUsernameField(usernameHint) {
+  const usernameField = document.getElementsByClassName(
+    'js-creator-signup-username',
+  )[0];
+  usernameField.value = usernameHint;
+}
+
+function showUsernameField() {
+  const usernameRow = document.getElementsByClassName(
+    'js-creator-signup-username-row',
+  )[0];
+  usernameRow.classList.remove('hidden');
+  focusUsernameInput(usernameRow);
+  hideHintRow();
+}
+
+function focusUsernameInput(usernameRow) {
+  // A timer with a count of 0 will run when the thread becomes idle
+  window.setTimeout(() => {
+    usernameRow.getElementsByTagName('input')[0].focus();
+  }, 0);
+}
+
+function hideHintRow() {
+  const hintRow = document.getElementsByClassName(
+    'js-creator-signup-username-hint-row',
+  )[0];
+  hintRow.classList.add('hidden');
+}
+
+function togglePasswordMask() {
+  visible = !visible;
+  togglePasswordType(visible);
+  toggleEyeIcons(visible);
+}
+
+function togglePasswordType(visible) {
+  const passwordType = visible ? 'text' : 'password';
+  passwordField.type = passwordType;
+}
+
+function toggleEyeIcons(visible) {
+  eyeOffIcon.classList.toggle('hidden', !visible);
+  eyeIcon.classList.toggle('hidden', visible);
+}
+
+let visible = false;
+const eyeIcon = document.getElementsByClassName('js-eye')[0];
+const eyeOffIcon = document.getElementsByClassName('js-eye-off')[0];
+const passwordField = document.getElementsByClassName('js-password')[0];
+const visibility = document.getElementsByClassName(
+  'js-creator-password-visibility',
```

Comment:
Refactor (blocking): Avoid prefixing CSS classes with `js-`. There is older code that does this, but it's a pattern to avoid.

Reply:
@nickytonline same as the `getElementsByClassName`, I think we need to establish and document the more accepted practices on the frontend. I prepend it with `-js` to show the dependencies on the js code, but should I be using ID's instead or what's the best practice going forward?

Reply:
Can you elaborate on that, because this is not just used in "old code"? Because having `js-` helps knowing that there's JS code associated with that element, in case someone is reading or refactoring, but I'm curious to know why it should be avoided, another thing probably to codify globally as a company

Reply:
I wasn't aware that we still use this pattern. There's a lot of code I haven't seen still I guess. 🙃  Is it only in the admin area?

Reply:
Just doing a quick search in the codebase, it's seems through articles, comments, badges and other files (so no not all admin related).  I'm not sure which of these are newer changes vs older. I know that I used this pattern recently in the admin sidebar as well.

Reply:
I can see how it can help someone discover that JS is interacting with an element, but doesn't seem necessary. I know it's still popular, but the quick Googling I did shows it's more related to vanilla JS and jQuery for separation of concerns.

If we're in Stimulus land, there are `data-*` attributes to indicate there is JS wired up, so a `js-*` class isn't even necessary, and in Preact, there are events, like `onClick`.

I'm guessing this pattern was used before we were using these libraries and just continued? Happy to have a wider discussion about this. Tagging @aitchiss as well to hear her thoughts on this.

Having said that, since it's already in several places, let's go ahead with this and continue the discussion in a roundtable or team post.

Reply:
Yeah I think this is a broader thing and there's a divergence of habits between different groups of folks on the team 😄  

From my POV, I wouldn't normally add a class name to get a handle to an individual item in JS, I'd use an ID. That's largely because it helps keeps class names related to styling - e.g. if I wanted a handle to all crayons dropdowns, I'd totally target them by `.crayons-dropdown`.  If I need a group of things that aren't related in style, I'd normally add a `data-attribute` so I can have more than one but again keep it separate to styling.

I'm not saying this is ^ is necessarily 'right', it's just what I do 😄  I agree @nickytonline that the `js` prefix puts me in mind of jquery. I don't think it does any harm, but we should suggest/find a unified approach so we can all get on the same page.

Reply:
> If we're in Stimulus land, there are data-* attributes to indicate there is JS wired up, so a js-* class isn't even necessary, and in Preact, there are event props, like onClick.

☝️ This makes sense. Like you said though `js-` is more related to vanilla js, which is what we're using in this PR.

Honestly, I'm a huge fan of the `js-` prefix as it gives it semantic meaning which makes it more readable and a lot easier to refactor because its so obvious what has associated js code. So whilst it doesn't seem 'necessary', I feel like it adds a lot of value. I know that using ID's are common as well, so I guess for our _vanilla javascript_ I'd like to weigh out the pro's and cons between ID's vs the js prefix, especially since we're haven't been commonly using stimulus for the non-admin code as yet. 


I agree that I don't really see the  value of the prefix in Stimulus or React land. 


Let's chat about this more in the round table or maybe a Forem.team post suggesting some  best practices for the Javascript code !